near-sdk/tests/store_performance_tests.rs

```diff
@@ -0,0 +1,268 @@
+        collection.to_string().to_lowercase()
+    );
+    let account_id: AccountId =
+        account_id.try_into().expect("could not convert dev account into AccountId");
+
+    account_id
+}
+async fn dev_generate(
+    worker: Arc<Worker<Sandbox>>,
+    collection: Collection,
+    seed: String,
+) -> anyhow::Result<(Account, Collection)> {
+    let id = random_account_id(collection, &seed);
+    let sk = SecretKey::from_seed(KeyType::ED25519, &seed);
+    let account = worker.create_tla(id.clone(), sk).await?;
+    Ok((account.into_result()?, collection))
+}
+#[tokio::test]
+#[allow(clippy::needless_range_loop)]
+// Note that different types of tests are executed sequentially, as the previous tests are populating
+// the data for the next ones.
+// Also, the tests for each collection are executed sequentially, as otherwise near-sandbox doesn't cope.
+async fn combined_test() -> anyhow::Result<()> {
+    let worker = Arc::new(near_workspaces::sandbox().await?);
+    let contract = worker.dev_deploy(include_bytes!("test-contracts/store/res/store.wasm")).await?;
+    let res = contract.call("new").max_gas().transact().await?;
+    assert!(res.is_success());
+    let contract_id = contract.id().clone();
+    let mut account_pool = Vec::new();
+
+    // Generate different accounts to avoid Nonce collisions when executing transactions in parallel.
+    // We could generate this per collection as well, but parallel execution for those proved to be
+    // too much for the sandbox, so we're keeping it simple.
+    for val in 0..=17 {
+        let (account, _) =
+            dev_generate(worker.clone(), Collection::TreeMap, val.to_string()).await?;
+        account_pool.push(account);
+    }
+
+    // insert
+    for (col, max_iterations) in Collection::iter().map(|col| match col {
+        // TreeMap performance is inferior to other collections.
+        Collection::TreeMap => (col, 15),
+        _ => (col, 16),
+    }) {
+        let account_pool = account_pool.clone();
+        let contract_id = contract_id.clone();
+        let mut total_gas: u64 = 0;
+        let mut futures = JoinSet::new();
+
+        for val in 0..max_iterations {
+            let account: Account = account_pool[val].clone();
+            let txn = account
+                .call(&contract_id.clone(), "exec")
+                .args_json((col, Op::Insert(val as u32)))
+                .transact();
+            futures.spawn(txn);
+        }
+        while let Some(res) = futures.join_next().await {
+            total_gas += res??.unwrap().total_gas_burnt.as_gas();
+        }
+        assert!(
+            total_gas < NearGas::from_tgas(100).as_gas(),
+            "performance regression: {}",
+            NearGas::from_gas(total_gas)
+        );
+        assert!(
+            total_gas > NearGas::from_tgas(90).as_gas(),
+            "not enough gas consumed: {}, adjust the number of iterations to spot regressions",
+            NearGas::from_gas(total_gas)
+        );
```

Comment:
Load so much data that each individual transaction execution consume 90..100TGas. No need to test total among several calls.

The hard limit for a single receipt is 300Tgas, so 90..100Tgas gives us room to catch the cases when it overflows